databroker/core.py

```diff
@@ -366,26 +361,19 @@ def documents_to_xarray(*, start_doc, stop_doc, descriptor_docs,
                 keys = scoped_data_keys
             for key, scoped_key in keys.items():
                 field_metadata = data_keys[key]
-                # Verify the actual ndim by looking at the data.
-                ndim = numpy.asarray(config['data'][key]).ndim
-                dims = None
-                if 'dims' in field_metadata:
-                    # As of this writing no Devices report dimension names ('dims')
-                    # but they could in the future.
-                    reported_ndim = len(field_metadata['dims'])
-                    if reported_ndim == ndim:
-                        dims = tuple(field_metadata['dims'])
-                    else:
-                        # TODO Warn
-                        ...
-                if dims is None:
-                    # Construct the same default dimension names xarray would.
-                    dims = tuple(f'dim_{next(dim_counter)}' for _ in range(ndim))
+                field_metadata = data_keys[key]
+                ndim = len(field_metadata['shape'])
+                # if the EventDescriptor doesn't provide names for the
+                # dimensions (it's optional) use the same default dimension
+                # names that xarray would.
+                dims = field_metadata.get(
+                    'dims',
+                    tuple(f'dim_{next(dim_counter)}' for _ in range(ndim)))
```

Comment:
I don't think the `dim_counter` is reset to the initial value anywhere, in the `for`-loop here one may get `dims = ('dim_100', 'dim_101', ...)`. Is it expected?

Reply:
I'm not sure about the answer for this. I am planning to replace this function in a future PR, which will get rid of this code.

Reply:
It used to be a conditional (`if dims is None: ...`) increment of the counter, but now it's unconditional, and the counter is not reset anywhere, which concerns me. I am not sure how serious the consequences of it, but I don't think we should let this potentially buggy code to appear in master even if the future PRs are going to replace the function. It's totally possible I am missing something.

@danielballan, do you have an idea about it?

Reply:
That's a good point. My intention here is that the counter would only be increment is `field_metadata` was missing the key `'dims'`, but as you point out the expression `tuple(f'dim_{next(dim_counter)}' for _ in range(ndim))` will be evaluated every time, whether `'dims'` is missing or not. Fix pushed.

Reply:
Thanks.